### PR TITLE
New Endpoint for PFAM domains

### LINKED ISSUE #60: Gene to PFAM domain mapping
Use Ensembl biomart to get Gene -> Transcript -> Translation -> PFAM domain mappings: http://www.ensembl.org/biomart/martview/

We need to generate a TSV file from this website and the use it to build our DB. See the screenshot for the required attributes when constructing the TSV file.

biomart_tsv_export

### PR DESCRIPTION
**Pull request by onursumer**
Fixes #60

Added a new endpoint for PFAM domains. The `PfamDomain` model currently reflects the biomart file, but we should probably extend it with additional domain information.

### CODE REVIEW

File: annotation/src/main/resources/application.properties.EXAMPLE [hunk 1]
```diff
@@ -15,6 +15,9 @@ mutationAssessor.url=http://mutationassessor.org/r3/?cm=var&var=VARIANT&frm=json
 # Comma separated list of isoform_overrides_resource:overrides_filename pairs.
 vep.isoform.overrides=uniprot:isoform_overrides_uniprot.txt,mskcc:isoform_overrides_at_mskcc.txt
 
+# Ensemble biomart PFAM filename
+ensemble.biomart.pfam=ensembl_biomart_pfam.txt
```
Comment (inodb) [inline annotation/src/main/resources/application.properties.EXAMPLE]: @onursumer should add this fail to the PR

Comment (onursumer) [reply annotation/src/main/resources/application.properties.EXAMPLE]: @inodb We don't have `isoform_overrides_uniprot.txt`, `isoform_overrides_at_mskcc.txt` in the repository either. But I guess it makes sense to maintain the biomart file in this repository.

Comment (inodb) [reply annotation/src/main/resources/application.properties.EXAMPLE]: @onursumer good point. I think we should actually store those and have some script to update/download them. Check if they're in sync. I'll create an issue

**Comment by inodb** [comment]
@onursumer could we change the property to default to the included biomart file path `ensembl.biomart.pfam`. Right now heroku fails because it can't find the biomart file. I think we should set up in such a way, such that it builds without doing any extra steps.

```
2017-09-26T09:14:31.995875+00:00 app[web.1]: java.lang.IllegalArgumentException: Could not resolve placeholder 'ensembl.biomart.pfam' in string value "${ensembl.biomart.pfam}"
2017-09-26T09:14:31.995876+00:00 app[web.1]:    at org.springframework.util.PropertyPlaceholderHelper.parseStringValue(PropertyPlaceholderHelper.java:174) ~[spring-core-4.2.4.RELEASE.jar!/:4.2.4.RELEASE]
```

**Comment by onursumer** [comment]
@inodb There was a typo in `application.properties.EXAMPLE`, now the heroku instance should work fine.

**Pull request merged**